Runtime/Extensions.cs

```diff
@@ -58,5 +59,68 @@ public static class Extensions
+        {
+            if (!serializableInterface.TryGetObject(out Object unityObject)) 
+            {
+                throw new System.Exception($"Cannot instantiate {serializableInterface} because it's has no reference of type UnityEngine.Object");
+            }
+
+            Object instantiatedObject = Object.Instantiate(unityObject);
+
+            return GetInterfaceReference<TInterface>(instantiatedObject);
+        }
+
+        public static TInterface Instantiate<TInterface>(this SerializableInterface<TInterface> serializableInterface, Transform parent) where TInterface : class
+        {
+            if (!serializableInterface.TryGetObject(out Object unityObject))
+            {
+                throw new System.Exception($"Cannot instantiate {serializableInterface} because it's has no reference of type UnityEngine.Object");
+            }
+
+            Object instantiatedObject = Object.Instantiate(unityObject, parent);
+
+            return GetInterfaceReference<TInterface>(instantiatedObject);
+        }        
+
+        public static TInterface Instantiate<TInterface>(this SerializableInterface<TInterface> serializableInterface, Vector3 position, Quaternion rotation) where TInterface : class
+        {
+            if (!serializableInterface.TryGetObject(out Object unityObject))
+            {
+                throw new System.Exception($"Cannot instantiate {serializableInterface} because it's has no reference of type UnityEngine.Object");
+            }
+
+            Object instantiatedObject = Object.Instantiate(unityObject, position, rotation);
+
+            return GetInterfaceReference<TInterface>(instantiatedObject);
+        }
+
+        public static TInterface Instantiate<TInterface>(this SerializableInterface<TInterface> serializableInterface, Vector3 position, Quaternion rotation, Transform parent) where TInterface : class
+        {
+            if (!serializableInterface.TryGetObject(out Object unityObject))
+            {
+                throw new System.Exception($"Cannot instantiate {serializableInterface} because it's has no reference of type UnityEngine.Object");
+            }
+
+            Object instantiatedObject = Object.Instantiate(unityObject, position, rotation, parent);
+
+            return GetInterfaceReference<TInterface>(instantiatedObject);
+        }
+
+        private static TInterface GetInterfaceReference<TInterface>(Object instantiatedObject) where TInterface : class
+        {
+            if (instantiatedObject is GameObject)
+            {
+                GameObject go = (GameObject)instantiatedObject;
+                if (!go.TryGetComponent(out TInterface tInterface)) return null;
+                return tInterface;
+            }
+            else
+            {
+                if (instantiatedObject is TInterface is false) return null;
+                return instantiatedObject as TInterface;
+            }
```

Comment:
This section can be shrunk down to the suggestion which improves the readability by a lot in my opinion

```suggestion
            if (instantiatedObject is GameObject gameObject)
                return gameObject.TryGetComponent(out TInterface component) ? component : null;

            return instantiatedObject as TInterface;
```